### DISCUSSION TITLE
Global Fetching State

### DISCUSSION THREAD

**Discussion by erichardson30** [urql-graphql/urql#2412] [Q&A]
Is there an equivalent of this https://react-query.tanstack.com/reference/useIsFetching in urql? 
I know each query gives you a `fetching` boolean ... but is there a way to determine if there are any queries pending?
(PS. using React)

**Comment by jdinartejesus** [comment]
I'm also interested in knowing such solution, couldn't find anyway a proper way.

Found this, but both links are gone.
https://spectrum.chat/urql/help/is-it-possible-to-get-a-global-level-fetching-status~252ada94-3203-4053-9390-fd53155cf877

**Comment by frederikhors** [comment]
I'm usign the below code which worked good until https://github.com/urql-graphql/urql/pull/3200#issuecomment-1569845321:

```ts
const pendingRequestsExchange: Exchange = ({ forward }) => {
  function before(op: Operation) {
    if (op.kind === 'subscription') return;
    if (op.kind && op.kind === 'teardown') {
      console.log('call deletePendingRequest');
      deletePendingRequest(op.key);
    } else {
      console.log('call addPendingRequest');
      addPendingRequest(op.key);
    }
  }
  function after(op: OperationResult) {
    deletePendingRequest(op.operation.key);
  }
  return (ops$) => {
    const forward$ = pipe(ops$, tap(before));
    return pipe(forward(forward$), tap(after));
  };
};
```

Did you find another way, guys?
